benches/lib.rs

```diff
@@ -0,0 +1,73 @@
+#![feature(test)]
+extern crate grcov;
+extern crate rustc_hash;
+extern crate test;
+
+use grcov::{CovResult, Function};
+use rustc_hash::FxHashMap;
+use test::{black_box, Bencher};
+
+#[bench]
+fn bench_lib_merge_results(b: &mut Bencher) {
+    let mut functions1: FxHashMap<String, Function> = FxHashMap::default();
+    functions1.insert(
+        "f1".to_string(),
+        Function {
+            start: 1,
+            executed: false,
+        },
+    );
+    functions1.insert(
+        "f2".to_string(),
+        Function {
+            start: 2,
+            executed: false,
+        },
+    );
+    let mut result = CovResult {
+        lines: [(1, 21), (2, 7), (7, 0)].iter().cloned().collect(),
+        branches: [
+            (1, vec![false, false]),
+            (2, vec![false, true]),
+            (4, vec![true]),
+        ]
+            .iter()
+            .cloned()
+            .collect(),
+        functions: functions1,
+    };
+
+    b.iter(|| {
+        let mut functions2: FxHashMap<String, Function> = FxHashMap::default();
```

Comment:
Couldn't you define functions2 and result2 outside b.iter?

Reply:
I was getting the error `cannot move out of captured outer variable in an `FnMut` closure` because I think merge_results takes ownership of result2? Should I move them outside and clone result?

Reply:
Yeah, try if cloning works.